app/models/bulkrax/oai_entry.rb

```diff
@@ -47,10 +47,9 @@ def build_metadata
     end
 
     def collections_created?
-      if sets.blank? || parser.collection_name != 'all'
-        self.collection_ids.length == 1
-      elsif parser.collection_name == 'all'
-        sets.length == self.collection_ids.length
+      
+      if parser.collection_name == 'all'
+        sets.blank? || (sets.exist? && sets.length == self.collection_ids.length)
```

Comment:
I know length is an alias, but size is the more common ruby array count method. I also know you didn’t change it from before, but if you are ok with it, I might swap them